charm.go

```diff
@@ -38,3 +40,29 @@ func ReadCharm(path string) (charm Charm, err error) {
 	}
 	return charm, nil
 }
+
+// SeriesToUse takes a specified series and a list of series supported by a
+// charm and returns the series which is relevant.
+// If the specified series is empty, the any default series defined by the
+// charm is used. Any specified series is validated against those supported
+// by the charm.
+func SeriesToUse(series string, supportedSeries []string) (string, error) {
```

Comment:
Rename the first parameter to "requestedSeries"? I think that would make its purpose a bit more obvious.